core/server/worker/src/main/java/alluxio/worker/block/UnderFileSystemBlockReader.java

```diff
@@ -315,14 +329,14 @@ private void updateBlockWriter(long offset) throws IOException {
     try {
       if (mBlockWriter == null && offset == 0 && !mBlockMeta.isNoCache()) {
         BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(mStorageTierAssoc.getAlias(0));
-        String blockPath = mLocalBlockStore
-            .createBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(), loc,
-                mInitialBlockSize).getPath();
-        mBlockWriter = new LocalFileBlockWriter(blockPath);
+        mLocalBlockStore.createBlock(mBlockMeta.getSessionId(), mBlockMeta.getBlockId(), loc,
+            mInitialBlockSize);
+        mBlockWriter = mLocalBlockStore.getBlockWriter(
+            mBlockMeta.getSessionId(), mBlockMeta.getBlockId());
       }
-    } catch (IOException | BlockAlreadyExistsException | WorkerOutOfSpaceException e) {
+    } catch (IOException | AlluxioException e) {
       // This can happen when there are concurrent UFS readers who are all trying to cache to block.
-      LOG.debug(
+      LOG.warn(
```

Comment:
could we `LOG.debug` on `BlockAlreadyExistsException` and `LOG.warn` on other exceptions?